src/nvim/terminal.c

```diff
@@ -1172,7 +1171,7 @@ static void redraw(bool restore_cursor)
     int off = is_focused(term) ? 0 : (curwin->w_p_rl ? 1 : -1);
     curwin->w_cursor.col = MAX(0, term->cursor.col + win_col_off(curwin) + off);
     curwin->w_cursor.coladd = 0;
-    setcursor();
+    mb_check_adjust_col(curwin);
```

Comment:
Don't we miss a UI cursor position update if we don't call `setcursor()` after `mb_check_adjust_col()`?